Don't try to normalise the channel if we don't have data for a product.

jgraham opened the pull request.

jgraham committed:
Don't try to normalise the channel if we don't have data for a product.

gsnedders reviewed (changes requested):
Please add tests so we don't regress `wpt run`. We should make sure `setup_wptrunner` can work with all known products.

jgraham committed:
Add a test for setup_wptrunner.

This actually requires that venv is avaiable but kindof does something
like the actual code.

gsnedders commented:
@jgraham any idea what caused this to fail CI? the tests that fail are all from @jugglinmike's #10998.

gsnedders commented on tools/wpt/tests/test_run.py:
```diff
@@ -16,3 +22,22 @@ def test_check_environ_fail(platform):
+
+
+@pytest.mark.parametrize("product", product_list)
+def test_setup_wptrunner(product):
```
This test seems to change some global state, making later tests fail.

`tox -e py27 -- --no-cov -k 'test_run_firefox or test_setup_wptrunner'` is sufficient to observe a failure; Travis has many more.

jgraham commented:
I made the requested changes some time ago. @gsnedders please re-review.

Hexcles reviewed (commented):
Found two bugs caused by this PR.

foolip commented:
Given the regression in https://github.com/web-platform-tests/wpt/pull/13567, what sort of test coverage are we missing here?
